include/boost/json/detail/impl/number.ipp

```diff
@@ -295,10 +330,15 @@ loop:
                     ++p;
                     pos_ = dig_;
                     n_.kind = kind::double_;
-                    continue;
+                    st_ = state::mantf;
+                    n_.u = m;
+                    goto loop;
                 }
                 if(*p == 'e' || *p == 'E')
                 {
+                    // 'E' implies '.' if not already encountered
```

Comment:
"if not already encountered" is superfluous, it is already spelled `if (pos_ < 0)`. This comment could be more succinct if written as `// treat 'E' as '.'`